src/main/java/pl/javagame/PasswordManager.java

```diff
@@ -1,35 +1,70 @@
 package pl.javagame;
 
 
-import java.util.ArrayList;
-import java.util.Random;
+import java.util.*;
 
 public class PasswordManager {
 
 
-    private ArrayList<String> passwords = new ArrayList<>();
+    private HashMap<String, Boolean> passwords = new HashMap<>();
+    // private ArrayList<String> copyPasswords = new ArrayList<>();
 
     public PasswordManager() {
-        passwords.add("Apetyt rośnie w miarę jedzenia");
-        passwords.add("Co dwie głowy to nie jedna");
-        passwords.add("Cwiczenie czyni mistrza");
-        passwords.add("Darowanemu koniowi w zęby się nie zagląda");
-        passwords.add("Diabeł tkwi w szczegółach");
-        passwords.add("Elektryka prąd nie tyka");
-
-        getRandomPassword();
+        passwords.put("Apetyt rośnie w miarę jedzenia", false);
+        passwords.put("Co dwie głowy to nie jedna", false);
+        passwords.put("Cwiczenie czyni mistrza", true);
+        passwords.put("Darowanemu koniowi w zęby się nie zagląda", false);
+        //    passwords.add("Diabeł tkwi w szczegółach");
+        //    passwords.add("Elektryka prąd nie tyka");
+
+        //    getRandomPassword();
     }
 
 
     public String getRandomPassword() {
-        int max = passwords.size(); // result=6
-        int i = (int) (Math.random() * max);
-        String randomPassword = passwords.get(i);
-        passwords.remove(i); // to avoid the same passwords
-        return randomPassword;
 
+        String Password = null;
+        String getKey;
+        Boolean getValue;
+
+        Integer max = passwords.keySet().toArray().length;// result=6
+        //  int random = new Random().nextInt(max);
+
+        //  System.out.println(Password);
+
+        Object randomPassword = passwords.keySet().toArray()[new Random().nextInt(max)];
+
+      //  randomPassword = checkigRandomPassword(randomPassword);
+
+        for (Map.Entry<String, Boolean> map : passwords.entrySet()) {
```

Comment:
Szczerze mówiąc nie rozumiem zasady działania tego fora.
Wyżej wylosowałeś hasło. 
Teraz trzebaby sprawdzić czy  passwords.get(randomPassword)==true, jeśli tak to je zwracamy, jeśli nie to losujemy kolejne.
Generalnie tak bym to widział
```
public String getRandomPassword() {
List<String> passwordsAsArray = passwords.keySet().toArray();

while(areThereUnusedPasswords()) {
String randomPassword  = passwordsAsArray[new Random().nextInt(max-1)];
if(passwords.get(randomPassword)) {
   password = randomPassword;
   passwords.set(randomPassword ,true)
   return password;
}
}
throw new IllegalStateException("No unused passwrods left");
}
private boolean areThereUnuseadPasswords() {
  for (Map.Entry<String, Boolean> entry : passwords.entrySet()) {
             if(entry.getValue()) return true;
  }
  return false;
}